nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/registry/flow/mapping/NiFiRegistryFlowMapperTest.java

```diff
@@ -0,0 +1,775 @@
+        final RemoteGroupPort remoteGroupOutputPort = prepareRemoteGroupPort(remoteProcessGroup);
+        when(remoteProcessGroup.getOutputPorts()).thenReturn(Sets.newHashSet(remoteGroupOutputPort));
+        return remoteProcessGroup;
+    }
+
+    private RemoteGroupPort prepareRemoteGroupPort(final RemoteProcessGroup remoteProcessGroup) {
+        final RemoteGroupPort remoteGroupPort = mock(RemoteGroupPort.class);
+        prepareComponentAuthorizable(remoteGroupPort, remoteProcessGroup.getIdentifier());
+        when(remoteGroupPort.getName()).thenReturn("remotePort" + (counter++));
+        when(remoteGroupPort.getRemoteProcessGroup()).thenReturn(remoteProcessGroup);
+        return remoteGroupPort;
+    }
+
+    private VersionControlInformation prepareVersionControlInfo() {
+        final VersionControlInformation versionControlInformation = mock(VersionControlInformation.class);
+        when(versionControlInformation.getRegistryIdentifier()).thenReturn(UUID.randomUUID().toString());
+        when(versionControlInformation.getBucketIdentifier()).thenReturn(UUID.randomUUID().toString());
+        when(versionControlInformation.getFlowIdentifier()).thenReturn(UUID.randomUUID().toString());
+        when(versionControlInformation.getVersion()).thenReturn(counter++);
+        return versionControlInformation;
+    }
+
+    private void prepareComponentAuthorizable(final ComponentAuthorizable authorizable, final String processGroupId) {
+        when(authorizable.getIdentifier()).thenReturn(UUID.randomUUID().toString());
+        when(authorizable.getProcessGroupIdentifier()).thenReturn(processGroupId);
+    }
+
+    private void preparePositionable(final Positionable positionable) {
+        when(positionable.getPosition()).thenReturn(new Position(counter++, counter++));
+    }
+
+    private void prepareConnectable(final Connectable connectable, final ConnectableType connectableType) {
+        when(connectable.getName()).thenReturn("connectable" + (counter++));
+        when(connectable.getConnectableType()).thenReturn(connectableType);
+    }
+
+    /**
+     * Verify the given VersionedProcessGroup was correctly mapped from the given ProcessGroup, including child groups
+     * when boolean indicator is true. If expectVersionControlInfo boolean indicator is true, the mapped group should
+     * only contain VersionControlInfo, otherwise it should instead contain its full contents.
+     */
+    private void verifyVersionedProcessGroup(final ProcessGroup processGroup, final VersionedProcessGroup versionedProcessGroup,
+                                             final boolean expectVersionControlInfo, final boolean verifyChildProcessGroups) {
+        final String expectedGroupIdentifier = flowMapper.getGroupId(processGroup.getProcessGroupIdentifier());
+
+        // verify process group fields
+        assertEquals(versionedProcessGroup.getName(), processGroup.getName());
```

Comment:
Not a big deal, but I think all of these asserts are passing the arguments backwards. The signature is `assertEquals(Object expected, Object actual)` but it looks like all of these assertions are made such that the actual value is passed before the expected value.

Reply:
Agreed, updating all